lib/scnnr/response.rb

```diff
@@ -39,7 +40,7 @@ def handle_recognition(recognition)
       end
     end
 
-    def handle_error
+    def raise_error_response!
```

Comment:
I have a question.
This method has `!` at the end of the name. That means the method raise some errors or means it works dangerously?
How do you decide the name?

Reply:
I've decided this name because raising an error is a side effect.
In addition, when the response is not `Net::HTTPSuccess` and the method is called, following process should not be executed.
So I intend to represent that the method always raises an error.

But the name without bang `raise_error_response` is similar to `raise SomeError, "..."`, so it may be enough.
What do you think about this?

Reply:
If we use a bang to represent "raising errors (or can cause side effects)", many of methods would have a bang.
I think we can determine it from a purpose of a method. In this case, the purpose of `raise_error_response!` is to occur a side effect.
From this point of view, option 2 would be closer.
Please note that this is just my opinion 😓

Reply:
Thank you, It makes sense and I want to avoid lots of methods have a bang too!

But to be honest, I think we should not use `!` except in the case of the situation that there are 2 methods, which one can cause side effects, and the other can not... (just like `create` and `create!` in Rails), because this is open source and it is too hard that everyone name methods correctly.

What do you think?

Reply:
I've investigated the concept of bangs. Is the convention you said from [Ruby core](https://docs.ruby-lang.org/en/trunk/syntax/methods_rdoc.html#label-Method+Names)?
If so and if you mean we should follow it, I agree with it 👍
If not so, please let me know the reference? Or it's OK that came from your thought.

Reply:
Thank you for the investigation! 
Honestly I didn't know the reference that you told me... but it seems like that this described all I want to say :)

Basically it was just my personal opinion, and referred to [here](https://bugs.ruby-lang.org/issues/5009) and [here](http://davidablack.net/dablog.html#2007/8/15/bang-methods-or-danger-will-rubyist).
I like the way of thinking that bang methods are dangerous version of non-bang methods, it is so simple and easy to keep for everyone :)

Reply:
I've finally understood! The links you wrote is very useful for me.
Of course, I follow general understanding if it exists.
Thank you for pointing out. I'll change it :)